pkg/controller/privatelink/actuator/gcpactuator/gcplinkactuator.go

```diff
@@ -145,45 +149,61 @@ func (a *GCPLinkActuator) Reconcile(cd *hivev1.ClusterDeployment, metadata *hive
+	var subnet *compute.Subnetwork
+	// Use a preexisting subnet if configured.
+	if subnetName := getServiceAttachmentSubnetExistingName(cd); subnetName != "" {
+		subnet, err = a.gcpClientSpoke.GetSubnet(subnetName, cd.Spec.Platform.GCP.Region, getServiceAttachmentSubnetExistingProject(cd))
 		if err != nil {
-			return reconcile.Result{}, errors.Wrap(err, "failed to update condition on cluster deployment")
+			return reconcile.Result{}, errors.Wrap(err, "failed to find the specified service attachment subnet")
 		}
-		return reconcile.Result{}, errors.Wrap(err, "failed to reconcile the Service Attachment Subnet")
-	}
-	if subnetModified {
-		err := conditions.SetReadyConditionWithRetry(*a.client, cd, corev1.ConditionFalse,
-			"ReconciledServiceAttachmentSubnet",
-			"reconciled the Service Attachment Subnet",
-			logger)
+		logger.WithField("subnetName", subnet.Name).Debug("reconciling Service Attachment Subnet: using existing subnet")
+	} else { // Otherwise create a new subnet
+		network, err := a.gcpClientSpoke.GetNetwork(metadata.InfraID + "-network")
 		if err != nil {
-			return reconcile.Result{}, errors.Wrap(err, "failed to update condition on cluster deployment")
+			return reconcile.Result{}, errors.Wrap(err, "failed to find the cluster Network")
 		}
-	}
-
-	logger.Debug("reconciling Service Attachment Firewall")
-	firewallModified, _, err := a.ensureServiceAttachmentFirewall(cd, metadata, network)
-	if err != nil {
-		logger.WithError(err).Error("failed to reconcile the Service Attachment Firewall")
 
-		err := conditions.SetErrConditionWithRetry(*a.client, cd, "ServiceAttachmentFirewallReconcileFailed", errors.New(controllerutils.ErrorScrub(err)), logger)
+		logger.Debug("reconciling Service Attachment Subnet")
+		subnetModified, newSubnet, err := a.ensureServiceAttachmentSubnet(cd, metadata, network)
 		if err != nil {
-			return reconcile.Result{}, errors.Wrap(err, "failed to update condition on cluster deployment")
+			logger.WithError(err).Error("failed to reconcile the Service Attachment Subnet")
+
+			err := conditions.SetErrConditionWithRetry(*a.client, cd, "ServiceAttachmentSubnetReconcileFailed", errors.New(controllerutils.ErrorScrub(err)), logger)
+			if err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "failed to update condition on cluster deployment")
+			}
+			return reconcile.Result{}, errors.Wrap(err, "failed to reconcile the Service Attachment Subnet")
 		}
-		return reconcile.Result{}, errors.Wrap(err, "failed to reconcile the Service Attachment Firwall")
-	}
-	if firewallModified {
-		err := conditions.SetReadyConditionWithRetry(*a.client, cd, corev1.ConditionFalse,
-			"ReconciledServiceAttachmentFirewall",
-			"reconciled the Service Attachment Firewall",
-			logger)
+		if subnetModified {
+			err := conditions.SetReadyConditionWithRetry(*a.client, cd, corev1.ConditionFalse,
+				"ReconciledServiceAttachmentSubnet",
+				"reconciled the Service Attachment Subnet",
+				logger)
```

Comment:
Should be we be tweaking this condition in the other code path? Like "Validated existing service attachment subnet" kind of thing?

Reply:
IMHO: no due to the fact that we are also not populating a value for service attachment subnet in the cd.status. We didn't really reconcile it so much as validate the input prior to using it.

Reply:
Ack. I was thinking more as a way to show progress, but I'm okay with it as is.